src/components/Table/GridTable.tsx

```diff
@@ -356,16 +377,50 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     const filteredRows = filterRows(api, columns, maybeSorted, filter);
     visitRows(filteredRows, 0, true);
 
-    return [headerRows, visibleDataRows, totalsRows, expandableHeaderRows, filteredRowIds];
-  }, [as, api, filter, maybeSorted, columns, style, rowStyles, sortOn, columnSizes, collapsedIds, getCount]);
+    // Check for any selected rows that are not displayed in the table because they don't match the current filter, or are no longer part of the `rows` prop.
+    // We persist these selected rows and hoist them to the top of the table.
+    if (sortedUnmatchedSelections.length) {
+      const hsrTuples: RowTuple<R>[] = sortedUnmatchedSelections.map((row, idx) => [
+        row,
+        // This change in `isLastUnmatchedSelectionRow` gives the bottom border, but also causes the rows to re-render. Would be nice to find another way to do it.
+        makeRowComponent(row, 1, true, idx === sortedUnmatchedSelections.length - 1),
+      ]);
+
+      // Add the children to the group row
+      unmatchedSelectionGroupRow.children = sortedUnmatchedSelections;
```

Comment:
Question: Hm, this `unmatchSelectionGroupRow` is a static import; can we avoid that? It seems like that would break if we had multiple GridTables rendering at the same time. I get that is probably rare.

Reply:
Ahh, good thought. I can update. I was trying to avoid having to build this `unmatchSelectionGroupRow` new every time, triggering unnecessary re-renders. But should be very small rerenders as this row is extremely minimal.